### PR TITLE
Recursive monitoring

### PR DESCRIPTION
**Pull request by danfruehauf**

### CODE REVIEW

File: icd-main.cpp [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #include <sys/stat.h>
 #include <cstdio>
 #include <cstring>
+#include <unistd.h>
```
Comment (hassonofer) [inline icd-main.cpp:31]: Doesn't see in use, maybe a mistake ?

Comment (danfruehauf) [inline icd-main.cpp:31]: Maybe I missed something, but it didn't compile for me. Complaining about `close` and `open` being undefined.

File: usertable.cpp [hunk 2]
```diff
@@ -250,6 +252,79 @@ UserTable::~UserTable()
...
+  syslog(LOG_DEBUG, "adding directory watch %s", sPath.c_str());
+
+  // warning only - permissions may change later
+  if (!(m_fSysTable || MayAccess(sPath, DONT_FOLLOW(pEntry->GetMask()))))
+    syslog(LOG_WARNING, "access denied on %s - events will be discarded silently", sPath.c_str());
+
+  try {
+    m_in.Add(pW);
+    m_map.insert(IWCE_MAP::value_type(pW, pEntry));
+  } catch (InotifyException e) {
```
Comment (hassonofer) [inline usertable.cpp]: Better to catch exceptions by reference for types that can be inherited.
Can save bad polymorphic casts.

Comment (danfruehauf) [inline usertable.cpp]: I should probably fix it, although this is a piece of code I've just moved.

**Pull request merged**